src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AbstractApiAction.java

```diff
@@ -606,10 +594,23 @@ protected boolean isValidRolesMapping(final RestChannel channel, final String ro
 		}
 
 		if (isReadOnly(rolesMappingConfiguration, role)) {
-			forbidden(channel, "Role '" + role + "' has read-only role-mapping.");
+			conflict(channel, "Role '" + role + "' has read-only role-mapping.");
 			return false;
 		}
 		return true;
 	}
 
+	boolean isResourceWriteProtected(final RestChannel channel, final SecurityDynamicConfiguration<?> configuration, final String name) {
```

Comment:
rename to `isWriteable()`

Reply:
This would mean the opposite of the behavior